pkg/secrethub/idp_link.go

```diff
@@ -0,0 +1,70 @@
+package secrethub
+
+import (
+	"github.com/secrethub/secrethub-go/internals/api"
+	"github.com/secrethub/secrethub-go/pkg/oauthorizer"
+)
+
+type IDPLinkService interface {
+	GCP() IDPLinkGCPService
+}
+
+func newIDPLinkService(client *Client) IDPLinkService {
+	return idpLinkService{
+		client: client,
+	}
+}
+
+type idpLinkService struct {
+	client *Client
+}
+
+func (i idpLinkService) GCP() IDPLinkGCPService {
+	return newIDPLinkGCPService(i.client)
+}
+
+type IDPLinkGCPService interface {
+	Create(namespace string, projectID string, authorizationCode string, redirectURI string) (*api.IdentityProviderLink, error)
+	List(namespace string) ([]*api.IdentityProviderLink, error)
```

Comment:
Should we return an iterator instead, so that we can easily introduce pagination later?

Reply:
I think we should. Since https://github.com/secrethub/secrethub-go/pull/163, all list functions have been deprecated in favor of iterators. I don't think it makes sense to introduce another list function now, just to deprecate it later.